devito/mpi/routines.py

```diff
@@ -898,6 +908,35 @@ class IrecvCall(Call):
     def __init__(self, parameters):
         super(IrecvCall, self).__init__('MPI_Irecv', parameters)
 
+
+class MPICall(Call):
+    pass
+
+
+class HaloUpdateCall(MPICall):
+    pass
+
+
+class HaloWaitCall(MPICall):
+    pass
+
+
+class RemainderCall(MPICall):
+    pass
+
+
+class MPIList(List):
+    pass
+
+
+class HaloUpdateList(MPIList):
+    pass
+
+
+class HaloWaitList(MPIList):
+    pass
+
+
```

Comment:
Why are all these virtual definitions necessary?

Reply:
Because they define special types that I need to search for in later stages of the compilation pipeline. For example, at some point I need to run an IET visitor catching all `HaloWaitCall` nodes, ie `FindNodes(HaloWaitCall).visit(...)`